### PR TITLE
Add greeyac protocol to IR Climate / HeatpumpIR

### PR DESCRIPTION
**Pull request by cmroche**
# What does this implement/fix? 

- Update HeatpumpIR dependency to tag 1.0.18 (https://github.com/ToniA/arduino-heatpumpir/compare/1.0.15...1.0.18)
- Add support for **greeyac** protocol
- When using **greeyac** and **sensor** is specified, send temperature to the device (Gree I-Feel)

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#1607

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [x] ESP8266

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
substitutions:
  device_name: thermystat
  friendly_name: Thermystat

# Homeassistant
sensor:
  - platform: homeassistant
    name: ${friendly_name} Temperature Input
    id: temperature_input
    entity_id: sensor.living_room_heater_current_temperature
    filters:
      - heartbeat: 120s

# Thermopump IR
remote_transmitter:
  pin: D0
  carrier_duty_percent: 50%

climate:
  - platform: heatpumpir
    name: ${friendly_name} IR
    protocol: greeyac
    sensor: temperature_input
    horizontal_default: auto
    vertical_default: auto
    min_temperature: 16
    max_temperature: 35
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/heatpumpir/climate.py [hunk 1]
```diff
@@ -111,4 +112,4 @@ def to_code(config):
     cg.add(var.set_max_temperature(config[CONF_MIN_TEMPERATURE]))
     cg.add(var.set_min_temperature(config[CONF_MAX_TEMPERATURE]))
 
-    cg.add_library("tonia/HeatpumpIR", "1.0.15")
+    cg.add_library("", "", "https://github.com/ToniA/arduino-heatpumpir.git#1.0.18")
```
Comment (cmroche) [inline esphome/components/heatpumpir/climate.py:117]: See the comments here to understand why this dependency is introduced this way: https://github.com/ToniA/arduino-heatpumpir/commit/0948c619d86407a4e50e8db2f3c193e0576c86fd

File: platformio.ini [hunk 2]
```diff
@@ -49,7 +49,7 @@ lib_deps =
     glmnet/Dsmr@0.5                                       ; dsmr
     rweather/Crypto@0.2.0                                 ; dsmr
     dudanov/MideaUART@1.1.8                               ; midea
-    tonia/HeatpumpIR@1.0.15                               ; heatpumpir
+    https://github.com/ToniA/arduino-heatpumpir.git#1.0.18                               ; heatpumpir
```
Comment (cmroche) [inline platformio.ini]: See the comment here to understand why this dependency is introduced this way: https://github.com/ToniA/arduino-heatpumpir/commit/0948c619d86407a4e50e8db2f3c193e0576c86fd

File: esphome/components/heatpumpir/climate.py (hunk 1 above)
Comment (oxan) [reply esphome/components/heatpumpir/climate.py:117]: Would be nice to add a comment to the source about this as well.

**Review by oxan — APPROVED** [review]

**Comment by oxan** [comment]
Thanks!

**Pull request merged**